### PR TITLE
Archetypes & Simulating Feature Assignments

### PR DESCRIPTION
**Pull request by Auz**
Adds way to test/simulate feature assignment and rules. You can add attributes to see what values the user will get, as well as save those simulated users to use through out the platform.

Screenshot 2023-09-22 at 8 18 05 AM

Screenshot 2023-09-22 at 8 18 21 AM

### CODE REVIEW

File: packages/back-end/src/routers/archetype/archetype.controller.ts [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+import { FeatureTestResult } from "../../../types/feature";
+import { evaluateFeature } from "../../services/features";
+import { getFeature } from "../../models/FeatureModel";
+import { promiseAllChunks } from "../../util/promise";
+
+// region GET /sample-users
+
+type GetArchetypeResponse = {
+  status: 200;
+  Archetype: ArchetypeInterface[];
```
Comment (bryce-fitzsimons) [inline packages/back-end/src/routers/archetype/archetype.controller.ts:31]: why is Archetype capitalized?

File: packages/back-end/src/routers/archetype/archetype.controller.ts [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+ */
+export const getArchetype = async (
+  req: AuthRequest,
+  res: Response<GetArchetypeResponse>
+) => {
+  const { org, userId } = getOrgFromReq(req);
+
+  req.checkPermissions("manageArchetype");
+
+  const Archetype = await getAllArchetype(org.id, userId);
```
Comment (bryce-fitzsimons) [inline packages/back-end/src/routers/archetype/archetype.controller.ts:48]: why capitalized?

Comment (Auz) [reply packages/back-end/src/routers/archetype/archetype.controller.ts:48]: Good catch - was a text replace problem

File: packages/back-end/src/models/ArchetypeModel.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  const newUser = await ArchetypeModel.create({
+    ...user,
+    id: uniqid("sam_"),
+    dateCreated: new Date(),
+    dateUpdated: new Date(),
+  });
+  return toInterface(newUser);
+}
+
+export async function getAllArchetype(
```
Comment (bryce-fitzsimons) [inline packages/back-end/src/models/ArchetypeModel.ts:68]: nit: should be `getAllArchetypes`

File: packages/back-end/src/models/ArchetypeModel.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+export async function deleteArchetypeById(id: string, organization: string) {
+  await ArchetypeModel.deleteOne({
+    id,
+    organization,
+  });
+}
+//
+// export function toArchetypeApiInterface(
```
Comment (bryce-fitzsimons) [inline packages/back-end/src/models/ArchetypeModel.ts:122]: still needed?

File: packages/back-end/src/types/Audit.ts [hunk 5]
```diff
@@ -7,6 +7,7 @@ export const EntityType = [
   "user",
   "organization",
   "savedGroup",
+  "Archetype",
```
Comment (bryce-fitzsimons) [inline packages/back-end/src/types/Audit.ts:10]: lowercase?

File: packages/back-end/types/audit.d.ts [hunk 6]
```diff
@@ -43,6 +43,9 @@ export type EventType =
   | "savedGroup.created"
   | "savedGroup.deleted"
   | "savedGroup.updated"
+  | "Archetype.created"
```
Comment (bryce-fitzsimons) [inline packages/back-end/types/audit.d.ts:46]: same

File: packages/front-end/components/Archetype/ArchetypeAttributesModal.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    name: string;
+    description: string;
+    attributes: string;
+    isPublic: boolean;
+  }>({
+    defaultValues: {
+      name: initialValues?.name || "",
+      description: initialValues?.description || "",
+      attributes: initialValues?.attributes || "",
+      isPublic: initialValues?.isPublic || true,
```
Comment (bryce-fitzsimons) [inline packages/front-end/components/Archetype/ArchetypeAttributesModal.tsx:27]: won't this always be true? Maybe use `??`

File: packages/back-end/src/models/ArchetypeModel.ts (hunk 3 above)
Comment (Auz) [reply packages/back-end/src/models/ArchetypeModel.ts:68]: fixed

File: packages/back-end/src/models/ArchetypeModel.ts (hunk 4 above)
Comment (Auz) [reply packages/back-end/src/models/ArchetypeModel.ts:122]: nope, removed

File: packages/back-end/src/types/Audit.ts (hunk 5 above)
Comment (Auz) [reply packages/back-end/src/types/Audit.ts:10]: fixed, thanks

**Review by bryce-fitzsimons — APPROVED** [review]

**Pull request merged**